packages/lit-html/src/private-ssr-support.ts

```diff
@@ -65,12 +68,28 @@ export const _$LH = {
       values: unknown[]
     ) => unknown
   ) => {
-    // TODO: This could be optimized such that `_$resolve` is only patched on
-    // the Directive base class once. This should reduce class shape changes and
-    // be slightly more optimal. The hazard is that we need to handle both the
-    // unminified and minified property keys.
     if (directiveClass.prototype._$resolve !== resolveOverrideFn) {
-      directiveClass.prototype._$resolve = resolveOverrideFn;
+      resolveName ??= directiveClass.prototype._$resolve.name;
+      let proto = directiveClass.prototype;
+      for (; proto !== Object.prototype; proto = Object.getPrototypeOf(proto)) {
```

Comment:
I personally prefer while loops over for loops w/o an initializer.

Reply:
Whoops, fixed with an initializer.

Reply:
Done!